src/ol/renderer/canvas/ImageLayer.js

```diff
@@ -80,13 +80,25 @@ CanvasImageLayerRenderer['create'] = function(mapRenderer, layer) {
       const candidate = /** @type {Object.<string, Function>} */ (candidates[i]);
       if (candidate !== CanvasImageLayerRenderer && candidate['handles'](RendererType.CANVAS, layer)) {
         renderer.setVectorRenderer(candidate['create'](mapRenderer, layer));
+        break;
```

Comment:
This looks like a good fix, but I'm curious when you are seeing more than one appropriate renderer being registered here.  I'm wondering if there are more bugs.

Reply:
The CanvasRenderer plugins are added in both Map.js and CanvasMap.js.

https://github.com/openlayers/openlayers/blob/ba17527d29a765460a295c8f0709e5115c31951a/src/ol/Map.js#L22-L28

https://github.com/openlayers/openlayers/blob/ba17527d29a765460a295c8f0709e5115c31951a/src/ol/CanvasMap.js#L18-L24

To my knowledge, the only other location where that could cause an issue was already handling it with a break.

https://github.com/openlayers/openlayers/blob/ba17527d29a765460a295c8f0709e5115c31951a/src/ol/renderer/Map.js#L210-L216

Reply:
Makes sense (though an application should not be importing both `Map` and `CanvasMap`).

Reply:
Ah, fair. That is an issue in the most recent released build, and I hadn't considered that it would be fixed with the move to es6 import/export.